superset/assets/javascripts/explore/AdhocMetric.js

```diff
@@ -0,0 +1,37 @@
+export default class AdhocMetric {
+  constructor(adhocMetric) {
+    this.column = adhocMetric.column;
+    this.aggregate = adhocMetric.aggregate;
+    this.hasCustomLabel = !!(adhocMetric.hasCustomLabel && adhocMetric.label);
+
+    if (this.hasCustomLabel) {
+      this.label = adhocMetric.label;
+    } else {
+      this.label = this.getDefaultLabel();
+    }
+
+    this.optionName = adhocMetric.optionName ||
+      `metric_${Math.random().toString(36).substring(2, 15)}_${Math.random().toString(36).substring(2, 15)}`;
+  }
+
+  getDefaultLabel() {
+    return `${this.aggregate || ''}(${(this.column && this.column.column_name) || ''})`;
```

Comment:
When would this.aggregate be null?

Reply:
It wouldn't be often, but in the case that someone clears the aggregate input in the edit popover, and then goes to edit the title, the default title they would see without this logic would be like `null(column_name)`, which seems like a bad user experience. Adding this logic would make the default title appear as `(column_name)`